File: commons-mis/src/main/java/io/github/cooperlyt/mis/work/impl/model/WorkOperatorModel.java
package io.github.cooperlyt.mis.work.impl.model;

import io.github.cooperlyt.mis.work.data.WorkOperator;
import lombok.*;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.domain.Persistable;
import org.springframework.data.relational.core.mapping.Column;
import org.springframework.data.relational.core.mapping.Table;

import java.time.LocalDateTime;

@EqualsAndHashCode(callSuper = true)
@Table("work_operator")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class WorkOperatorModel extends WorkOperator implements Persistable<String> {

  public WorkOperatorModel(WorkOperator operator, String id) {
    super(operator);
    this.id = id;
  }

  @Builder(builderMethodName = "operatorBuilder")
  public WorkOperatorModel(long workId, String empId, String empName,
                           OperatorType type, Long orgId, String orgName, String id) {
    super(workId, empId, empName, type, orgId, orgName);
    this.id = id;
  }

  @Id
  @Column("task_id")
  private String id;

  @CreatedDate
  @Override
  public LocalDateTime getWorkTime() {
    return super.getWorkTime();
  }

  @Override
  public boolean isNew() {
    return true;
  }
}

File: commons-mis/src/main/java/io/github/cooperlyt/mis/work/impl/model/WorkOperatorModel.kt
package io.github.cooperlyt.mis.work.impl.model

import io.github.cooperlyt.mis.work.data.WorkActionType
import io.github.cooperlyt.mis.work.data.WorkOperator
import io.github.cooperlyt.mis.work.data.WorkOperatorSupplier
import org.springframework.data.annotation.Id
import org.springframework.data.domain.Persistable
import org.springframework.data.relational.core.mapping.Table
import java.time.LocalDateTime

@Table("work_operator")
data class WorkOperatorModel(
  @Id
  val taskId: String,
  val workId: Long,
  override val userId: String,
  override val userName: String,
  override val type: WorkActionType,

  override val workTime: LocalDateTime,

  override val orgName: String?,
  override val corpInfoId: Long?,
  override val employeeInfoId: Long?
): WorkOperator, Persistable<String> {

  constructor(taskId: String, workId: Long, type: WorkActionType, operator: WorkOperatorSupplier): this(
    taskId,
    workId,
    operator.userId,
    operator.userName,
    type,
    LocalDateTime.now(),
    operator.orgName,
    operator.corpInfoId,
    operator.employeeInfoId
  )

  override fun getId(): String? = taskId

  override fun isNew(): Boolean = true
}
